src/main/javascript/crypto/src/eas/EasAttestedObject.ts

```diff
@@ -0,0 +1,112 @@
+import {EasTicketAttestation, TicketSchema} from "./EasTicketAttestation";
+import {EASIdentifierAttestation} from "./EASIdentifierAttestation";
+import {AttestationCrypto} from "../libs/AttestationCrypto";
+import {ProofOfExponentInterface} from "../libs/ProofOfExponentInterface";
+import {hexStringToUint8, logger, uint8tohex} from "../libs/utils";
+import {AsnParser, AsnProp, AsnPropTypes, AsnSerializer} from "@peculiar/asn1-schema";
+import {KeyPair, KeysArray} from "../libs/KeyPair";
+import {UsageProofOfExponent} from "../libs/UsageProofOfExponent";
+import {DEBUGLEVEL} from "../config";
+
+export class EasUseToken {
+
+	@AsnProp({ type: AsnPropTypes.Any })
+	public ticketAttestation: Uint8Array;
+
+	@AsnProp({ type: AsnPropTypes.Any })
+	public idAttestation: Uint8Array;
+
+	@AsnProp({ type: AsnPropTypes.Any })
+	public proof: Uint8Array;
+}
+
+export class EasAttestedObject {
+
+	private crypto = new AttestationCrypto();
+
+	private ticketAttestation: EasTicketAttestation;
+	private ticketSecret: bigint;
+	private identifierAttestation: EASIdentifierAttestation;
+	private identifierSecret: bigint;
+
+	private pok: ProofOfExponentInterface;
+
+	create(
+		ticketAttestation: EasTicketAttestation,
+		ticketSecret: bigint,
+		identifierAttestation: EASIdentifierAttestation,
+		identifierSecret: bigint){
+
+		this.ticketAttestation = ticketAttestation;
+		this.ticketSecret = ticketSecret;
+		this.identifierAttestation = identifierAttestation;
+		this.identifierSecret = identifierSecret;
+
+		this.pok = this.makeProof();
+	}
+
+	private makeProof(): ProofOfExponentInterface {
+
+		let attCom: Uint8Array = this.identifierAttestation.getCommitment();
+		let objCom: Uint8Array = this.ticketAttestation.getCommitment();
+		let pok: ProofOfExponentInterface = this.crypto.computeEqualityProof(uint8tohex(attCom), uint8tohex(objCom), this.identifierSecret, this.ticketSecret);
+
+		if (!this.crypto.verifyEqualityProof(attCom, objCom, pok)) {
+			throw new Error("The redeem proof did not verify");
+		}
+		return pok;
+	}
+
+	getEncoded(){
+
+		const useToken = new EasUseToken();
+		useToken.ticketAttestation = new Uint8Array(this.ticketAttestation.getAsnEncoded());
```

Comment:
@micwallace , no need to use ASN1 encoding in this case, better stay it solid and encode everything with defaultAbiCoder, like internal data encoded
in this case we dont need to use separate lib to decode ASN. it can save GAS in contract execution and save GAS when we deploy smaller contract.

Reply:
Yep I agree. Keep in mind the contracts will require updating too. Can you help with that if I go ahead and make this change?

Reply:
@oleggrib I've added an ABI encoded version. This is used by default but I've kept the ASN version around until the next major version.